protoc.mk

```diff
@@ -0,0 +1,28 @@
+# This Makefile makes it easy for the language bindings
+# to download and install the protoc binary.
+
+PROTOC_OS := $(shell uname -s)
+ifeq (Darwin,$(PROTOC_OS))
+PROTOC_OS := osx
+endif
+
+PROTOC_ARCH := $(shell uname -m)
```

Comment:
Instead of that, let's allow users to pass in the value directory. `uname` does not work on windows.